storyscript/Cli.py

```diff
@@ -79,8 +81,13 @@ def parse(path, debug, ebnf, raw, ignore, lower, preview):
         Parses stories, producing the abstract syntax tree.
         """
         try:
-            trees = App.parse(path, ignored_path=ignore, ebnf=ebnf,
-                              lower=lower, features=preview)
+            if not os.isatty(0):
+                story = sys.stdin.read()
+                trees = App.parse(story=story, ebnf=ebnf,
+                                  lower=lower, features=preview)
+            else:
+                trees = App.parse(path=path, ignored_path=ignore, ebnf=ebnf,
+                                  lower=lower, features=preview)
```

Comment:
```suggestion
args = {}
if not os.isatty(0):
    args['story'] = sys.stdin.read()
else:
   args['path'] = path
App.parse(**args, <other args>)
```

Or of course add the other args already to `args`. The advantage is that you can now create a method of the first part and can reuse it for the lines below, e.g.

```
args = build_input(path)
```